keps/sig-node/5526-pod-level-resource-managers/README.md

```diff
@@ -0,0 +1,1616 @@
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone /
+release*.
+
+-   [ ](R) Enhancement issue in release milestone, which links to KEP dir in
+    [kubernetes/enhancements](not the initial KEP PR)
+-   [ ](R) KEP approvers have approved the KEP status as `implementable`
+-   [ ](R) Design details are appropriately documented
+-   [ ](R) Test plan is in place, giving consideration to SIG Architecture and
+    SIG Testing input (including test refactors)
+    -   [ ] e2e Tests for all Beta API Operations (endpoints)
+    -   [ ](R) Ensure GA e2e tests meet requirements for
+        [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+    -   [ ](R) Minimum Two Week Window for GA e2e tests to prove flake free
+-   [ ](R) Graduation criteria is in place
+    -   [ ](R)
+        [all GA Endpoints](https://github.com/kubernetes/community/pull/1806)
+        must be hit by
+        [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+        within one minor version of promotion to GA
+-   [ ](R) Production readiness review completed
+-   [ ](R) Production readiness review approved
+-   [ ] "Implementation History" section is up-to-date for milestone
+-   [ ] User-facing documentation has been created in [kubernetes/website], for
+    publication to [kubernetes.io]
+-   [ ] Supporting documentation—e.g., additional design documents, links to
+    mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+-->
+
+This KEP proposes extending the Kubelet's Topology, CPU, and Memory Managers to
+support pod-level resource specifications. Currently, these managers operate on
+a per-container basis. This enhancement will enable them to make NUMA alignment
+and resource allocation decisions for a pod as a single unit, based on
+`pod.spec.resources`. This change introduces a more flexible and powerful
```

Comment:
This seems incorrect. There's a way to do NUMA alignment for a pod as a single unit today. And even with the new approach the allocation will still be done per container if I am not wrong. We are not planning to use pod-level cpuset. The only thing that changes is we are using pod-level resources values for hint generation, instead of aggregating the container-level values when scope is set to pod. I will read through the KEP to get back on what changes when scope is set to container.

Reply:
You are right, the underlying logic will still produce the allocation by each container. I meant to say enabling the alignment and allocation using `pod.spec.resources`. I will update the phrasing to be more clear about the actual changes, so there is no confusion.

Reply:
Done, wdyt?